xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-server/src/main/java/org/xwiki/rest/internal/exceptions/ExceptionExceptionMapper.java

```diff
@@ -0,0 +1,81 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.rest.internal.exceptions;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
+
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.rest.XWikiRestComponent;
+
+import static javax.ws.rs.core.MediaType.TEXT_PLAIN_TYPE;
+import static org.apache.commons.lang3.exception.ExceptionUtils.getStackTrace;
+
+/**
+ * Exception mapper for all {@link Exception} exceptions. Allowing to log all exceptions that are not inheriting from
+ * {@link RuntimeException} or already mapped by another exception mapper.
+ * <p>
+ * The goal is to be able to have access to the stacktrace of the exceptions, instead of a simple http 500 error and a
+ * {@code "No ExceptionMapper was found, but must be found"} message (see
+ * <a href="https://github.com/restlet/restlet-framework-java/issues/1070">
+ * https://github.com/restlet/restlet-framework-java/issues/1070</a>).
+ *
+ * @version $Id$
+ * @since 13.8RC1
+ * @since 13.7.1
+ * @since 13.4.4
+ */
+@Component
+@Named("org.xwiki.rest.internal.exceptions.ExceptionExceptionMapper")
+@Provider
+@Singleton
+public class ExceptionExceptionMapper implements ExceptionMapper<Exception>, XWikiRestComponent
+{
+    @Inject
+    private Logger logger;
+
+    @Override
+    public Response toResponse(Exception cause)
+    {
+        // This ExceptionMapper captures too many exception types, because we can't distinguish checked and unchecked
+        // exceptions using generic types. If the mapped type is RuntimeException (unchecked exception), JaxRsProviders
+        // would re-throw the exception without mapping it (or it would have been handled by another mapper instead of
+        // this one), we reproduce this behavior to stay as close as possible to the default implementation (note:
+        // RuntimeException stacktraces are logged by default).
+        if (cause instanceof RuntimeException) {
+            throw (RuntimeException) cause;
+        }
+
+        // Logs the unmapped exception, with its stacktrace.
+        this.logger.error("A REST endpoint failed with an unmapped exception.", cause);
+
+        // Returns an error response to the client, with some detail about the exception in the response.
+        String message = String.format("No ExceptionMapper was found for [%s].%n%s", cause, getStackTrace(cause));
```

Comment:
`%n` seems to mean "platform-specific line separator" but you are sending a content to an HTTP client here, and you are not going to read it on the server, so there is no need to be server system dependent. I guess a simple `\n` would do.

I'm wondering if it would not be better to produce this response content in a template. We have tools to produce stack traces (`#displayException` for example) that we use in various places.